lib/autoproj/ops/build.rb

```diff
@@ -84,7 +89,42 @@ def force_build_packages(selected_packages, all_enabled_packages)
             def build_packages(all_enabled_packages, options = Hash.new)
                 Autobuild.do_rebuild = false
                 Autobuild.do_forced_build = false
-                Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                begin
+                    Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                ensure
+                    build_report(all_enabled_packages) if @report_dir
+                end
+            end
+
+            REPORT_BASENAME = "build_report.json"
+
+            # The path to the report file
+            #
+            # @return [String,nil] the path, or nil if the report should not
+            #    be generated
+            def report_path
+                File.join(@report_dir, REPORT_BASENAME) if @report_dir
+            end
+
+            def build_report(package_list)
+                FileUtils.mkdir_p @report_dir
+
+                packages = package_list.map do |pkg_name|
+                    pkg = Autobuild::Package[pkg_name]
```

Comment:
I think you'll have to at least pass a Manifest object to the class in order to avoid accessing Autobuild::Package